torch_geometric/nn/encoding.py

```diff
@@ -94,3 +96,182 @@ def forward(self, x: Tensor) -> Tensor:
 
     def __repr__(self) -> str:
         return f'{self.__class__.__name__}({self.out_channels})'
+
+
+# TODO: Generalize the module when needed
+class _MLPMixer(torch.nn.Module):
+    """1-layer MLP-mixer for GraphMixer.
+
+    Args:
+        num_tokens (int): The number of tokens (patches) in each sample.
+        in_channels (int): Input channels.
+        out_channels (int): Output channels.
+    """
+    def __init__(
+        self,
+        num_tokens: int,
+        in_channels: int,
+        out_channels: int,
+    ) -> None:
+        super().__init__()
+        # token mixing
+        self.token_layer_norm = torch.nn.LayerNorm((in_channels, ))
+        self.token_lin_1 = torch.nn.Linear(num_tokens, num_tokens // 2)
+        self.token_lin_2 = torch.nn.Linear(num_tokens // 2, num_tokens)
+
+        # channel mixing
+        self.channel_layer_norm = torch.nn.LayerNorm((in_channels, ))
+        self.channel_lin_1 = torch.nn.Linear(in_channels, 4 * in_channels)
+        self.channel_lin_2 = torch.nn.Linear(4 * in_channels, in_channels)
+
+        # head
+        self.head_layer_norm = torch.nn.LayerNorm((in_channels, ))
+        self.head_lin = torch.nn.Linear(in_channels, out_channels)
+
+    def forward(self, x: Tensor) -> Tensor:
+        """
+        Args:
+            x (torch.Tensor): Tensor of size ``[N, num_tokens, in_channels]``
+
+        Returns:
+            Tensor of size ``[N, out_channels]``
+        """
+        # token mixing
+        h = self.token_layer_norm(x).mT
+        h = self.token_lin_1(h)
+        h = torch.nn.functional.gelu(h)
+        h = torch.nn.functional.dropout(h, p=0.5, training=self.training)
+        h = self.token_lin_2(h)
+        h = torch.nn.functional.dropout(h, p=0.5, training=self.training)
+        h_token = h.mT + x
+
+        # channel mixing
+        h = self.channel_layer_norm(h_token)
+        h = self.channel_lin_1(h)
+        h = torch.nn.functional.gelu(h)
+        h = torch.nn.functional.dropout(h, p=0.5, training=self.training)
+        h = self.channel_lin_2(h)
+        h = torch.nn.functional.dropout(h, p=0.5, training=self.training)
+        h_channel = h + h_token
+
+        # head
+        h_channel = self.head_layer_norm(h_channel)
+        t = torch.mean(h_channel, dim=1)
+        return self.head_lin(t)
```

Comment:
nit: Are `self.head_layer_norm` and `self.head_in` needed? They are not mentioned in the paper.

Reply:
I didn't have this originally, but I added this to match my implementation with theirs:  https://github.com/CongWeilin/GraphMixer/blob/bbdd9923706a02d0619dfd00ef7880911f003a65/model.py#L239

Reply:
Got it. thanks for clarifying. Then let's leave it.